### PR TITLE
Adapt to the move of metadata-code from `datalad` to `datalad-deprecated`

### PR DESCRIPTION
**Pull request by christian-monch**
This PR adapts the parts of `datalad_neuroimaging` that used the old metadata code, i.e. the code contained in `datalad.metadata`, to the new location of this code, i.e. `datalad_deprecated.metadata`.

### CODE REVIEW

**Comment by christian-monch** [comment]
Remaining errors should be fixed after `datalad-deprecated` is released with PR https://github.com/datalad/datalad-deprecated/pull/71

**Comment by yarikoptic** [comment]
Thank you ! I have merged and released deprecated 0.2.5 , you might like to add versioning to depends

File: setup.cfg [hunk 1]
```diff
@@ -17,6 +17,7 @@ zip_safe = False
 python_requires = >= 3.7
 install_requires =
     datalad >= 0.16.7
+    datalad_deprecated
```
Comment (bpoldrack) [inline setup.cfg:20]:
```suggestion
    datalad_deprecated >= 0.2.5
```

**Comment by bpoldrack** [comment]
Thx, @christian-monch!
Merging ...

**Pull request merged**